server/resp/src/main/java/org/infinispan/server/resp/commands/generic/SCAN.java

```diff
@@ -0,0 +1,140 @@
+
+   public SCAN() {
+      super(-2, 1, -1, 1);
+   }
+
+   @Override
+   public CompletionStage<RespRequestHandler> perform(Resp3Handler handler,
+                                                      ChannelHandlerContext ctx,
+                                                      List<byte[]> arguments) {
+      int argc = arguments.size();
+      if (argc < 1) {
+         WRONG_ARGUMENT_COUNT_BICONSUMER.accept("scan", handler.allocatorToUse());
+         return handler.myStage();
+      }
+      String filterConverterFactory = null;
+      List<byte[]> filterConverterParams = null;
+      int count = -1;
+      RespTypes type = null;
+      if (argc > 1) {
+         for(int i = 1; i < argc; i++) {
+            byte[] arg = arguments.get(i);
+            if (Arrays.compare(MATCH, arg) == 0) {
+               if (++i >= argc) {
+                  Consumers.SYNTAX_ERROR_BICONSUMER.accept(null, handler.allocatorToUse());
+                  return handler.myStage();
+               } else {
+                  filterConverterFactory = GlobMatchFilterConverterFactory.class.getName();
+                  filterConverterParams = Collections.singletonList(arguments.get(i));
+               }
+            } else if (Arrays.compare(COUNT, arg) == 0) {
+               if (++i >= argc) {
+                  Consumers.SYNTAX_ERROR_BICONSUMER.accept(null, handler.allocatorToUse());
+                  return handler.myStage();
+               } else {
+                  try {
+                     count = Integer.parseInt(new String(arguments.get(i), StandardCharsets.US_ASCII));
+                  } catch (NumberFormatException e) {
+                     Consumers.VALUE_NOT_INTEGER_BICONSUMER.accept(null, handler.allocatorToUse());
+                     return handler.myStage();
+                  }
+               }
+            } else if (Arrays.compare(TYPE, arg) == 0) {
+               if (++i >= argc) {
+                  Consumers.SYNTAX_ERROR_BICONSUMER.accept(null, handler.allocatorToUse());
+                  return handler.myStage();
+               } else {
+                  try {
+                     type = RespTypes.valueOf(new String(arguments.get(i), StandardCharsets.US_ASCII));
+                  } catch (IllegalArgumentException e) {
+                     type = RespTypes.unknown;
+                  }
+               }
+            }
+         }
+      }
+
+      DefaultIterationManager iterationManager = handler.respServer().getIterationManager();
+      String cursor = new String(arguments.get(0), StandardCharsets.US_ASCII);
+      if ("0".equals(cursor)) {
+         // New iterator
+         if (count < 0) {
+            count = DEFAULT_COUNT;
+         }
+         IterationState iterationState = iterationManager.start(handler.cache(), null, filterConverterFactory, filterConverterParams, MediaType.APPLICATION_OCTET_STREAM, count, false, DeliveryGuarantee.EXACTLY_ONCE);
```

Comment:
I believe it could be `AT_LEAST_ONCE`?

Reply:
Correct, this is what we want eventually. Unfortunately, the argument isn't currently used, but might as well keep the code the way it should be.